bugsnag-android-core/src/main/java/com/bugsnag/android/EventPayload.kt

```diff
@@ -30,6 +30,58 @@ class EventPayload : JsonStream.Streamable {
         this.notifier = notifier
     }
 
+    fun getErrorTypes(): String? {
+        return when {
+            event != null -> getErrorTypesFromStackframes(event)
+            eventFile != null -> getErrorTypesFromFilename(eventFile)
+            else -> null
+        }
+    }
+
+    private fun getErrorTypesFromStackframes(event: Event): String? {
+        val errorTypes = event.errors.mapNotNull(Error::getType).toSet()
+        val frameOverrideTypes = event.errors
+            .map { it.stacktrace }
+            .flatMap { it.mapNotNull(Stackframe::type) }
+
+        val distinctTypes = errorTypes.plus(frameOverrideTypes)
+        return serializeErrorTypeHeader(distinctTypes)
+    }
+
+    private fun getErrorTypesFromFilename(eventFile: File): String? {
+        val name = eventFile.name
+        val end = name.lastIndexOf("_", name.lastIndexOf("_") - 1)
+        val start = name.lastIndexOf("_", end - 1) + 1
+
+        if (start < end) {
+            val errorTypes = name.substring(start, end)
+            val validValues = ErrorType.values().map { it.desc }
+
+            // validate that this only contains valid error type info
+            val valid = errorTypes.split(",").all { errorType ->
+                validValues.contains(errorType)
+            }
+            if (valid) {
+                return errorTypes
+            }
+        }
+        return null
+    }
```

Comment:
This doesn't feel like the correct place to put this logic - it's separated from the code in `EventStore` that writes the filename and requires that `EventStore` constructs an `EventPayload` just to call this method.

Can we abstract it into some sort of helper class that deals with encoding and decoding the filename so that it can be accessed easily in both contexts and the encoding and decoding (including `getApiKeyFromFilename`) is maintained together.

Reply:
I've extracted this into the `EventFilenameInfo` file, which is responsible for encoding/decoding event information from filenames. It additionally extracted several functions that were previously in the `EventStore`.